Documentation of Application Process for Samsung TV Seller Office

UgaitzEtxebarria opened the pull request:
PR to visualize and organize the documentation of the process for Samsung TV Seller Office

UgaitzEtxebarria committed:
Created README.md

UgaitzEtxebarria committed:
Added samsung ppt info in READMEs

Jyasapara committed:
Add guide for Language Options

UgaitzEtxebarria commented:
Thank you very much, I don't have access to the app on tizen, just desktop or web, then every screenshot is welcome

uxdesignerhector commented:
Official Jellifyn Icons and backgrounds for Tizen can be found on this repo https://github.com/jellyfin/jellyfin-ux under the next path https://github.com/jellyfin/jellyfin-ux/tree/master/branding/tizen. I have been looking to it and they are generating them with Inkscape and command line tools so we don't need to recreate them. 

They seem to be following rules specified at [Samsung's Guide for App icons](https://developer.samsung.com/smarttv/design/app-icons-and-screenshots.html#app-icon ) the only thing we must take into account is that the naming should be `logo_jellfyfin_fg.png` and `logo_jellyfin_bg.png` as per official Samsung rules ( [see here](https://developer.samsung.com/smarttv/design/app-icons-and-screenshots.html#app-icon) ),  right now those images are named `logo_fg.png` and `logo_bg.png`

Here are the routes for foreground and background images:
- https://github.com/jellyfin/jellyfin-ux/blob/master/branding/tizen/logo_bg.png
- https://github.com/jellyfin/jellyfin-ux/blob/master/branding/tizen/logo_fg.png

UgaitzEtxebarria committed:
Merge pull request #1 from JasperJuergensen/master

Add guide for Language Options

UgaitzEtxebarria committed:
Added resume for Application UI Description in general README

UgaitzEtxebarria committed:
Added icon requirements and restructure the folders

UgaitzEtxebarria committed:
Changes in image requirements

UgaitzEtxebarria committed:
Foreground app image done

Jyasapara committed:
Add missing 2015 logo and screenshots to App Information

Screenshots are taken using a web browser using the [demo
instance](https://demo.jellyfin.org/stable). They should look more or
less the same as on the tv.

UgaitzEtxebarria committed:
Merge pull request #2 from JasperJuergensen/app-info

Add missing 2015 logo and screenshots to App Information

UgaitzEtxebarria committed:
Changed icons from 32 bits to 24 bit png

UgaitzEtxebarria committed:
Changed README for structure it

UgaitzEtxebarria committed:
Added the file size requirement

J-C-B committed:
Update README.md

Added Remote Control mappings

UgaitzEtxebarria committed:
Merge pull request #3 from J-C-B/patch-1

Update README.md

UgaitzEtxebarria committed:
Update README.md

Mark Remoto Control Keys as marked

J-C-B committed:
Update README.md - added gslides for colab

J-C-B committed:
Update README.md - colab for slides

UgaitzEtxebarria committed:
Merge pull request #4 from J-C-B/patch-2

Update README.md - added gslides for colab

UgaitzEtxebarria committed:
Merge pull request #5 from J-C-B/patch-3

Update README.md - colab for slides

J-C-B committed:
fixed typos and added examples

J-C-B committed:
added req3 screenshots

J-C-B committed:
fixed order numbering

J-C-B committed:
added req 2 screenshots and flows

J-C-B committed:
removed folders

J-C-B commented:
I have added about 30 flows and described screenshots that should be sufficient - happy to add more is needed

UgaitzEtxebarria committed:
Merge pull request #6 from J-C-B/master

Screen Shots, flow diagrams for req 2 and 3

J-C-B committed:
Update README.md

added colab gslides

J-C-B committed:
completed requirements and ui structure diagram

J-C-B committed:
Merge branch 'UgaitzEtxebarria:master' into master

J-C-B commented:
As far as i can tell I have created all the required screenshots and flows and they are sitting in the forks pull requests - I assume the "project" will want to own the listing in the samsung store - who will take it from here for final submission?

UgaitzEtxebarria committed:
Merge pull request #7 from J-C-B/patch-4

Update README.md

UgaitzEtxebarria committed:
Merge pull request #8 from J-C-B/master

all diagrams complete

UgaitzEtxebarria committed:
Update README.md

Checked te screenshot that are correct

UgaitzEtxebarria commented:
I think that @anthonylavado will submit the info because he is the owner of the [main task](https://github.com/jellyfin/jellyfin-tizen/issues/222) and he is part of Jellyfin organization. In addition, he says that he has the ID of samsung App. If someone knows something more just add the info :)

UgaitzEtxebarria committed:
Added Additional requirements and changes needed

UgaitzEtxebarria committed:
renaming

UgaitzEtxebarria committed:
Update README.md

Format fixed

dmitrylyzo commented:
FYI, to get TV-like screenshots, you can use "emulation" of the device in Chrome DevTools.
You need a new device:
```
Name: Samsung TV Tizen 8
Resolution: 1920x1080
Device pixel ratio: 1 or empty
User agent string: Mozilla/5.0 (SMART-TV; Linux; Tizen 8.0) AppleWebKit/537.36 (KHTML, like Gecko) SamsungBrowser/7.0 Chrome/108.0.5359.1 TV Safari/537.36
Type: Desktop
```

UgaitzEtxebarria committed:
Update README.md

Deleted tasks

UgaitzEtxebarria committed:
Update README.md

Removed tasks

J-C-B committed:
Updated image and table with demo creds

UgaitzEtxebarria committed:
Merge pull request #12 from J-C-B/master

Updated image and table with demo creds

J-C-B committed:
Reduced file sizes for app info image

J-C-B committed:
Reduced file sizes for app info

J-C-B committed:
Merge branch 'master' of https://github.com/J-C-B/jellyfin-tizen

J-C-B committed:
updated checklist

J-C-B committed:
removed extra items

J-C-B committed:
changed png to jpg to reduce file size <500kb

screenshots can be jpg based on this link https://developer.samsung.com/tv-seller-office/application-publication-process.html - so png not required :)

J-C-B committed:
delete pngs

screenshots can be jpg based on this link https://developer.samsung.com/tv-seller-office/application-publication-process.html - so png not required

J-C-B committed:
updated readme to show JPG selection reason

J-C-B committed:
Update README.md

UgaitzEtxebarria committed:
Merge pull request #13 from J-C-B/master

reduced app info image sizes <500kb

Mitchell-Faas commented:
@UgaitzEtxebarria Since you mentioned that the version is ready for review, here are some gaps I'm still seeing:

**UI Diagram**
- Not alls UI paths are represented, especially around settings. I'm not sure they'll accept grouping everything under one "settings" item. (think display, playback, subtitles, and controls)

**Usecases**
- Menu browsing is not covered in the use cases section
- Login flow lacks some detail on side effects, such as how the password gets reset.
- Media playback portion in the usecases is incomplete. Some of the slides are not part of usecases, but should instead be part of the "menus and functions" section (describing the UI)

**Menus and Functions**
This section is significantly lacking in its description and completeness. Some examples of missing things are: Home screen, media carousels, browsing screen, filters, cast crew, recommendation grid.
All screenshots must also be explicitly labeled with numbers or text, and each label must be described.


Keep in mind that all descriptions must be detailed and comprehensive enough for a QA tester to conclusively validate the correct functioning of the app. At this point it would fail on lack of usecase descriptions (especially in media playback) and incomplete UI description.

nielsvanvelzen commented:
The logo assets are using our old font, you should update them to the latest version [SVG](https://jellyfin.org/images/logo.svg)

J-C-B commented:
Thanks for the feedback everyone - @anthonylavado - as you will be taking this forward - where do you feel we have landed? are you happy to take it forward as is? or specific things you would like improved?

UgaitzEtxebarria committed:
Moved slices from Use Cases to Menus and Functions

anthonylavado commented:
Hi folks,
Thank you very much for this. I will have some time to review it this weekend and then I can see where we'll go next 🙂

anthonylavado commented:
Alright, did a quick overview of the text. Seems to be all good, but I'll go through it further while comparing to the Samsung documentation just to make sure everything is landed correctly.

For the three Google Slides presentations I saw, I made copies to my own Google Drive account. We'll probably export them as .ODP files and keep them available for reference. Once I upload those, I'll remove the explicit links to Google in those readme files, and point to the uploaded version instead.

I'll take care of updating the logos to use the new font.

Once I've updated the images and uploaded the slides/ODP, then I'll push to this branch, convert it to "ready" status, and merge.

dmitrylyzo commented:
1. Are we going to store "Documentation" in this repo (we don't store it for webOS :man_shrugging: )?
Currently we will have to exclude `Documentation/*` from the build. I was thinking of refactoring the build process to generate a separate `dist` folder from which the app will be packaged. So it won't be necessary in the future.

2. The screenshots are taken in the Desktop display mode. We should take them in the TV display mode. https://github.com/jellyfin/jellyfin-tizen/pull/307#issuecomment-2476031425

3. We need to clean up the git history of PR. Squash/rearrange commits.

J-C-B commented:
> 1. Are we going to store "Documentation" in this repo (we don't store it for webOS 🤷‍♂️ )?
>        Currently we will have to exclude `Documentation/*` from the build. I was thinking of refactoring the build process to generate a separate `dist` folder from which the app will be packaged. So it won't be necessary in the future.
> 
>     2. Some screenshots are taken in the Desktop display mode. We should take them in the TV display mode. [Documentation of Application Process for Samsung TV Seller Office #307 (comment)](https://github.com/jellyfin/jellyfin-tizen/pull/307#issuecomment-2476031425)
> 
>     3. We need to clean up the git history of PR. Squash/rearrange commits.

Sorry, not sure if you are asking me or Anthony, but here are my thoughts

1, It would be good to keep the documentation public and available should it need to be reused in the future - perhaps we could move it to the wiki? https://github.com/jellyfin/jellyfin-tizen/wiki
2, I Did setup the emulation as suggested about half way through - but found very little difference in the actual UI, personally I think they are a close enough likeness to the TV for the process. However if they are a must I can recapture them.
3, I am not sure how to do this, happy to take guidance on this process if I need to do something.

UgaitzEtxebarria commented:
> Alright, did a quick overview of the text. Seems to be all good, but I'll go through it further while comparing to the Samsung documentation just to make sure everything is landed correctly.
> 
> For the three Google Slides presentations I saw, I made copies to my own Google Drive account. We'll probably export them as .ODP files and keep them available for reference. Once I upload those, I'll remove the explicit links to Google in those readme files, and point to the uploaded version instead.
> 
> I'll take care of updating the logos to use the new font.
> 
> Once I've updated the images and uploaded the slides/ODP, then I'll push to this branch, convert it to "ready" status, and merge.

I agree with @dmitrylyzo, this MR was created to store the documentation, not to been merged and add unnecessary info for who needa just compile the code. I specified some options here https://github.com/jellyfin/jellyfin-tizen/issues/222#issuecomment-2464259263, and was selected store it on a branch. Web can chance if its a better way to manage the documentation.

Apart from that, I am creating new issues with the problems that people says, if you look any issue that needs to be add, just add it.

Thanks for all contribution

J-C-B commented:
@dmitrylyzo and @anthonylavado - I have created the new screenshots in the TV ui - I am just unclear how to provide them based on the discssion above and dont want to keep re-creating documentation.

I looked at the main jellyfin project and they have `docs` in a docs folder - https://github.com/jellyfin/jellyfin.org/tree/master/docs - why could we not do the same? it would be consistant with the main project then.

If we use the wiki feature of github we will still need an images directory hosted somewhere to be able to show the screenshots in the wiki via url in any case.

I am happy to help and contribute in the way you would like me to, but would like a solid direction to go in before I spend the time to do it.

J-C-B committed:
named to docs

J-C-B committed:
added link

J-C-B committed:
Update README.md

J-C-B committed:
removed spaces from folders

J-C-B committed:
Update README.md

J-C-B committed:
pages test

J-C-B committed:
Update README.md

J-C-B committed:
Update README.md

J-C-B committed:
index

J-C-B committed:
pages again

J-C-B committed:
Create _config.yml

J-C-B committed:
removed spaces

J-C-B committed:
removed more folder spaces

J-C-B committed:
formatting

J-C-B committed:
added other sections

J-C-B committed:
new screenshots

J-C-B committed:
corrected naming and made links

J-C-B committed:
Update README.md

J-C-B committed:
Update README.md

J-C-B committed:
added tv screenshots

J-C-B committed:
Update README.md

J-C-B committed:
Update README.md

J-C-B committed:
Update README.md

UgaitzEtxebarria committed:
Merge pull request #22 from J-C-B/master

updated screenshots and readme linking

J-C-B commented:
Hi @anthonylavado  - i have made updates a couple of weeks ago - are you able to review? is the application submitted? thanks!

anthonylavado commented:
I'd like to thank everyone involved for their work on this. I am going to push the files to a separate repository to avoid clogging up this one with anything that would need to be ignored/excluded from any development processes.

IMO - it's not a bad idea to move it to a repository like our Jellyfin-meta, as it really is higher level administrative stuff, and only rarely applies. My other projects (work and personal) have slowed down for now, and some work for improving our publishing flows are now complete.

J-C-B commented:
> I'd like to thank everyone involved for their work on this. I am going to push the files to a separate repository to avoid clogging up this one with anything that would need to be ignored/excluded from any development processes.
> 
> IMO - it's not a bad idea to move it to a repository like our Jellyfin-meta, as it really is higher level administrative stuff, and only rarely applies. My other projects (work and personal) have slowed down for now, and some work for improving our publishing flows are now complete.

OK awesome, please let me know if i can help further and where the files end up. Appreciate the time you put in too!

I know there are a few requests for this in places - will you link back to here for the new file location so people can follow along certification progress without haveing to ask?

UgaitzEtxebarria commented:
@anthonylavado do you make some progress on this?
I think that all the files are migrated to new repo, If you give me your confirmation I will close the PR

J-C-B commented:
> @anthonylavado do you make some progress on this? I think that all the files are migrated to new repo, If you give me your confirmation I will close the PR

yes its all moved over so this PR can be closed from that perspective - so unless this is handy for tracking i think we can close this one

Nightcro commented:
@anthonylavado any update on this?

J-C-B commented:
> Are these the last steps before it will be officially published?

The new PR is here - https://github.com/jellyfin/jellyfin-meta/pull/84 - @anthonylavado is carrying out the process with Samsung, but i have not seen and details on progrogress for a few months. - i suggest folloing the new PR for updates
